web/controllers/pair_controller.ex

```diff
@@ -63,14 +57,37 @@ defmodule Pairmotron.PairController do
   end
 
   defp generate_pairs(year, week) do
-    User.active_users
+    users = User.active_users
       |> order_by(:id)
       |> Repo.all
+
+    pairs = fetch_pairs(year, week)
+      |> Repo.preload(:users)
+
+    determination = PairBuilder.determify(pairs, users)
+
+    determination.dead_pairs
+      |> Enum.map(fn(p) -> Repo.delete! p end)
```

Comment:
Could this use `Repo.delete_all` instead of of `Repo.delete!` on each pair? `delete_all` would be one trip to the db for all pairs whereas Repo.delete! is a trip to the db for each pair.